applications/tari_base_node/src/grpc/base_node_grpc_server.rs

```diff
@@ -1123,14 +1123,27 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
 
     async fn get_constants(
         &self,
-        _request: Request<tari_rpc::Empty>,
+        request: Request<tari_rpc::BlockHeight>,
     ) -> Result<Response<tari_rpc::ConsensusConstants>, Status> {
         debug!(target: LOG_TARGET, "Incoming GRPC request for GetConstants",);
         debug!(target: LOG_TARGET, "Sending GetConstants response to client");
-        // TODO: Switch to request height
-        Ok(Response::new(
-            self.network.create_consensus_constants().pop().unwrap().into(),
-        ))
+
+        let block_height = request.into_inner().block_height;
+        let network_consensus_constants = self.network.create_consensus_constants();
+
+        let mut consensus_constants = &network_consensus_constants[0];
```

Comment:
ConsensusManager already gets the appropriate constants for the height
```suggestion
let consensus_manager = ConsensusManager::builder(self.network.as_network()).build();
    
        let mut consensus_constants = consensus_manager.consensus_constants(block_height);
```

Reply:
Cool ! I was unaware of `ConsensusManager`. Simplifies it a lot ! :)

Reply:
Addressed